### DISCUSSION TITLE
Delayed notifications

### DISCUSSION THREAD

**Discussion by i3laze** [RavenSystem/esp-homekit-devices#1748] [New Features]
Sometimes one need to trigger a series of actions or several switches sequentially.
Current workaround for complex notifications is to create some dummy inching switches and hook a notification (or HomeKit.app automation) when these dummies go off.

Few examples:
- (Auto)Close garage door 15 sec after motion detected.
- Turn on some GPIO then another one [after some delay](https://github.com/RavenSystem/esp-homekit-devices/discussions/1743).

I suggest to expand intra-HAA notifications with a delay, like “md”:[[a, v, d]], where _d_ is a delay in seconds for action to be triggered.

Feel free to discuss is there any usefulness and provide few more examples.

**Comment by RavenSystem** [comment]
None of your examples needs a special delay feature.

- (Auto)Close garage door 15 sec after motion detected must be done with an inching declared in motion sensor.
- Turn on some GPIO then another one after some delay must be done directly using inching in binary outputs action array.

**Comment by i3laze** [comment]
Please elaborate on a second example:
“I need to turn on GPIO1 now and GPIO2 three seconds later.”
How this is currently possible without a dummy inching switch or turning GPIO1 off?
Invoke g:2, v:0, i:3 while GPIO2 still low??

**Reply by RavenSystem** [reply]
Yes. `"r":[{"g":1,"v":1},{"g":2,"i":3}]`

**Reply by 3uperduck** [reply]
But when I apply this on a double lock mechanism function which is removed last year can't work well.
Delay before second lock mechanism unlock is set to 2 seconds.
Here is code

{"c":{"l":13,"b":[{"g":0,"t":5}]},
"a": [
    {
      "t":4,
      "i":32,
      "0":{"r":[{"g":5,"v":1,"i":0.5},{"m":[{"A":2,"V":0}],"i":2}]},
      "1":{"r":[{"g":5}]},
      "b":[{"g":0,"p":0,"i":1,"t":0}],
      "es": [
        {
          "t":4,
 	    "i":1,
    	    "0":{"r":[{"g":12,"v":1,"i":0.5}]},
    	    "1":{"r":[{"g":12}]},
 	    "b":[{"g":9,"p":0,"i":1,"t":0}]
        }
      ]
    }
  ]
}

**Comment by i3laze** [comment]
You can’t (yet) declare notifications with inching:
m":[{"A":2,"V":0}],"i":2 seems not valid 

Instead just trigger GPIO12 with inching from first lock and also send a notification to change state.

**Reply by 3uperduck** [reply]
yeah, but have no idea how to do

**Reply by i3laze** [reply]
That’s a question for @RavenSystem 
Trigger inching accessory with a delay is hardcore.

I personally would add an extra dummy switch with i:2 and notify it on from lock1, then when switch off notify lock2. That is the scenario I’m asking here to improve.
